packages/compute/src/vm.js

```diff
@@ -809,6 +829,139 @@ VM.prototype.request = function(reqOpts, callback) {
   });
 };
 
+/**
+ * Poll `getMetadata` to check the VM's status. This runs a loop to ping
+ * the API on an interval.
+ *
+ * Note: This method is automatically called when a `waitFor()` call
+ * is made.
+ *
+ * @private
+ */
+VM.prototype.startPolling_ = function() {
+  var self = this;
+
+  if (!this.hasActiveWaiters) {
+    return;
+  }
+
+  this.getMetadata(function(err, metadata) {
+    var now = new Date();
+    for (var index in self.waiters) {
+      if (err) {
+        self.waiters[index].callback(err, null);
+        self.waiters.splice(index, 1);
+      } else if (metadata && metadata.status === self.waiters[index].status) {
+        self.waiters[index].callback(null, metadata);
+        self.waiters.splice(index, 1);
+      } else {
+        if ((now.getTime() / 1000) - self.waiters[index].startTime >=
+              self.waiters[index].timeout) {
+          self.waiters[index].callback({
+            name: 'WaitForTimeout'
+          },null);
+          self.waiters.splice(index, 1);
+        }
+      }
+    }
+    if (self.waiters.length <= 0) {
+      self.hasActiveWaiters = false;
+    } else {
+      setTimeout(self.startPolling_.bind(self), WAIT_FOR_POLLING_INTERVAL);
+    }
+  });
+};
+
+/**
+ * This function will callback when the VM is in the specified state.
+ *
+ * Will time out after the specified time (default: 300 seconds).
+ *
+ * @param {string} status - The status to wait for. This can be:
+ *  - "PROVISIONING"
+ *  - "STAGING"
+ *  - "RUNNING"
+ *  - "STOPPING"
+ *  - "SUSPENDING"
+ *  - "SUSPENDED"
+ *  - "TERMINATED"
+ * @param {object=} options - Configuration object.
+ * @param {number} options.timeout - The number of seconds to wait until
```

Comment:
At the point that you are taking a `timeout` option, you could also take an `interval` option, and default it to `2000`.